src/app/forms/custom-sections/adr-tank-statement-un-number-edit/adr-tank-statement-un-number-edit.component.html

```diff
@@ -1,40 +1,44 @@
-<ng-container *ngIf="control && formArray.controls[0]">
-  <app-field-error-message
-    [error]="!formArray.controls[0].value && control.invalid && (control.touched || submitted) ? control.meta.customErrorMessage : ''"
-  ></app-field-error-message>
+<ng-container *ngIf="control && formArray.controls.length">
+  <app-field-error-message [error]="control.touched ? control.getError('custom')?.message : ''"></app-field-error-message>
 
-  <ng-container *ngIf="formArray.controls.length === 1">
+  <ng-container *ngIf="formArray.controls.length === 1; else multiple">
     <app-text-input
-      [id]="control.meta.name"
-      [name]="formArray.controls[0].meta.name"
-      [label]="formArray.controls[0].meta.label + ' 1'"
-      [customId]="formArray.controls[0].meta.customId"
+      [id]="'UN_number_1'"
+      [name]="'UN number 1'"
+      [label]="'UN number 1'"
+      [customId]="'UN number 1'"
       [formControl]="formArray.controls[0]"
+      [minlength]="1"
+      [maxlength]="1500"
       (blur)="formArray.controls[0].markAsTouched(); control.markAsTouched()"
     ></app-text-input>
   </ng-container>
 
-  <!-- Additional UN numbers -->
-  <table class="govuk-table" *ngIf="formArray.controls.length > 1">
-    <tbody class="govuk-table__body">
-      <ng-container *ngFor="let child of formArray.controls; let i = index">
-        <tr>
-          <td class="govuk-table__cell">
-            <app-text-input
-              [name]="child.meta.name"
-              [label]="(child.meta.label ?? '') + ' ' + (i + 1)"
-              [customId]="child.meta.customId"
-              [formControl]="child"
-              (blur)="child.markAsTouched(); control.markAsTouched()"
-            ></app-text-input>
-          </td>
-          <td class="govuk-table__cell align-middle">
-            <a class="govuk-link" role="button" href="javascript:void(0)" (click)="removeControl(i)">Remove</a>
-          </td>
-        </tr>
-      </ng-container>
-    </tbody>
-  </table>
+  <ng-template #multiple>
+    <table class="govuk-table">
+      <tbody class="govuk-table__body">
+        <ng-container *ngFor="let child of formArray.controls; let i = index">
+          <tr>
+            <td class="govuk-table__cell">
+              <app-text-input
+                [id]="'UN_number_' + (i + 1)"
+                [name]="'UN number ' + (i + 1)"
+                [label]="'UN number ' + (i + 1)"
+                [customId]="'UN number ' + (i + 1)"
+                [formControl]="child"
+                [minlength]="1"
+                [maxlength]="1500"
+                (blur)="child.markAsTouched(); control.markAsTouched()"
+              ></app-text-input>
+            </td>
+            <td class="govuk-table__cell align-middle">
+              <a class="govuk-link" role="button" href="javascript:void(0)" (click)="removeControl(i)">Remove</a>
```

Comment:
No to javascript void please, recommend looking up why it's bad and using an alternative.

Reply:
I disagree with the online reasons for why this is bad - they seem to be complaining about how IE handles this and alternatives like href="#" is even worse because it scrolls to the top of the page, but I agree that semantically this is not ideal. This was kind of a hack because the govuk-link style doesn't apply properly for anchor tags when no href is given, but will change to via css as that's probably better.

Reply:
in angular can you not just {{ return false }} or something?

Reply:
I also agree href=# is bad haha

Reply:
in angular can you not just {{ return false }} or something? - no clue - changed it to use css as that was what other people did in other places.